.github/workflows/python-package.yml

```diff
@@ -0,0 +1,38 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python package
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.9", "3.10", "3.11"]
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        pip install --upgrade pip
+        pip install flake8 pytest sympy
+    - name: Install package
+      run: |
+        pip install -r requirements.txt
+        pip install -r test-requirements.txt
+        pip install .
+    - name: Test with pytest
+      run: |
+        pytest
```

Comment:
i'm totally not an expert for GitHub actions but it seems there is a problem with `sympy` not installed. Is it in the `test-requirements.txt` too ? 

You can also give a look to the GitHub Actions docs for Python. It seems quite good: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python